src/index.ts

```diff
@@ -10,6 +10,13 @@ const equivalents = [
   'quotes'
 ] as const
 
+function fromEntries<T> (iterable: Array<[string, T]>): { [key: string]: T } {
+  return [...iterable].reduce<{ [key: string]: T }>((obj, [key, val]) => {
+    obj[key] = val
+    return obj
+  }, {})
+}
+
```

Comment:
What I had trouble with is having multiple different possible types for the index signature. Here only `string` is used.

This might be losing some type information in comparison with how the `in` operator is used [here](https://github.com/microsoft/TypeScript/blob/5321dcb09f4d37768662afcfc9ffbe1f6b24d0c9/lib/lib.es2019.object.d.ts#L28). But probably good enough, right?

Reply:
Yeah, that is what my thinking was. We just need it to be good enough. We are not planning on using any other index types but `string` and this is not exported nor a standalone piece of the library, just a small built-in util 😄